File: core/src/com/ntnu/game/states/play/PlayState_1.java
package com.ntnu.game.states.play;

import com.badlogic.gdx.graphics.g2d.SpriteBatch;
import com.ntnu.game.HelicopterGame;
import com.ntnu.game.sprites.Helicopter;
import com.ntnu.game.states.GameStateManager;
import com.ntnu.game.states.State;

public class PlayState_1 extends State {

    private Helicopter helicopter;

    public PlayState_1(final GameStateManager gsm) {
        super(gsm);
        helicopter = new Helicopter(50, 100);
    }

    @Override
    public void update(float dt) {
        helicopter.update();
    }

    @Override
    public void render(SpriteBatch sb) {
        sb.begin();
        sb.draw(helicopter.getHelicopter(),
                helicopter.getPosition().x,
                helicopter.getPosition().y,
                HelicopterGame.HELICOPTER_WIDTH,
                HelicopterGame.HELICOPTER_HEIGHT);
        sb.end();
    }

    @Override
    public void dispose() {
        helicopter.dispose();
    }
}

File: core/src/com/ntnu/game/states/play/PlayState1.kt
package com.ntnu.game.states.play

import com.badlogic.gdx.graphics.g2d.SpriteBatch
import com.badlogic.gdx.math.Vector2
import com.ntnu.game.GameLogic
import com.ntnu.game.sprites.Helicopter
import com.ntnu.game.states.BaseState
import kotlin.random.Random

class PlayState1 : BaseState() {

    private val helicopter: Helicopter = Helicopter(Vector2(Random.nextFloat() * 10f + 5, Random.nextFloat() * 13f + 8),
            Vector2(5f, 5f))

    private val logic: GameLogic = GameLogic()

    init {
        logic.addHelicopter(helicopter)
        createBackButton()
    }

    override fun update(dt: Float) {
        logic.update(dt)
    }

    override fun render(sb: SpriteBatch) {
        super.render(sb)
        logic.render(sb)
    }

    override fun dispose() {
    }
}
